x-pack/elastic-agent/pkg/remote/client.go

```diff
@@ -277,18 +275,20 @@ func prefixRequestFactory(URL string) requestFunc {
 }
 
 // makeTransport create a transport object based on the TLS configuration.
-func makeTransport(scheme string, timeout time.Duration, tls *tlscommon.Config) (http.RoundTripper, error) {
-	dialer := transport.NetDialer(timeout)
-	if scheme == "http" {
-		return &http.Transport{Dial: dialer.Dial}, nil
+func makeTransport(cfg Config, scheme string) (http.RoundTripper, error) {
+	opts := []httpcommon.TransportOption{
+		httpcommon.WithAPMHTTPInstrumentation(),
 	}
-	tlsConfig, err := tlscommon.LoadTLSConfig(tls)
-	if err != nil {
-		return nil, errors.Wrap(err, "invalid TLS configuration")
-	}
-	tlsDialer, err := transport.TLSDialerH2(dialer, tlsConfig, timeout)
-	if err != nil {
-		return nil, errors.Wrap(err, "fail to create TLS dialer")
+
+	// Connect to fleet server via HTTP2 only if no proxy is configured.
+	// The HTTP2 only transport will ignore HTTP_PROXY, HTTPS_PPROXY, and NO_PROXY
+	// environment variables.
+	http2Only := scheme == "https"
+	if http2Only && cfg.Transport.Proxy.URL == nil {
+		opts = append(opts, httpcommon.WithHTTP2Only(true))
+	} else {
+		opts = append(opts, httpcommon.WithForceAttemptHTTP2(true))
```

Comment:
Why not just have it be force attempt http2 always? There is no requirement that it has to be http2, but we prefer http2 for the benefits of duplex requests over same TLS/connection.

Reply:
+1 on ForceAttempt only. I assumed that with your recent change you don't want to fallback to http 1 in case we can't go http2.